### ISSUE TITLE
Workflow / Project Structure docs

### ISSUE THREAD

**Issue by danbohea** [hashbangcode/vlad#120]
I've now read, edited and re-read `docs/usage/workflow.md` several times today and 3 major things still stand out to me that I feel need to be corrected:

1. My own workflow isn't listed (_waah_).
2. We're still recommending that people update Vlad by downloading from GitHub and then carefully overwrite certain files & directories.
3. It's still quite a confusing document.

Proposed solutions (same order as above):

1. This is entirely my fault for having a different workflow in the first place as well as promising to add it to the doc and then not getting round to it. I will fix this and in turn this _should_ help us with point 2...

2. Vlad's been tweaked enough now for this kind of update process to not be necessary anymore (unless the _local_ directory structure is arranged a certain way and it's out of the question to rearrange it). I suggest that we promote a Git based workflow for updating Vlad first and foremost and then go on to *lightly* cover this kind of optional manual process further down the doc (with appropriate warnings). I think we should be encouraging best practice here for all sorts of splendid reasons, not least to avoid people hitting the issue queue after accidentally borking their installation by manually copy/overwriting files with a dash of human error :D

3. The implications surrounding .gitignore, updating Vlad, repositories, Git submodules, data loss, custom roles, where to store Vlad settings, where to put docroot, what to call docroot and Christ knows what else make this a pretty tricky document to get right :) I'd like a pop at rewriting the whole thing starting with an overview of the options available through Vlad, going on to cover an "ideal" (and why) setup and then gradually covering use cases that drift further away from "ideal". Use cases would still be numbered for easy reference both in and outside of the doc. This may even end up as a few pages rather than one monster page for easier digestion and the author's own threadbare mental capacity.

I'm still writing `<cough>`_starting_`</cough>` my Vlad session for DrupalCamp Brighton at the mo so there's definitely some potential overlap here that we could hopefully benefit from. Thoughts?

**Comment by danbohea** [comment]
OK, I've uploaded a first draft of what I'm thinking here:

https://github.com/hashbangcode/vlad/blob/dev/vlad/docs/usage/project_structure.md

This is all just one giant single page still for now, that's partly because I'm still scratching my head over some MkDocs concepts:

1. Can you do nesting deeper than one level in MkDocs?
2. If not, is it worth breaking "Project Structure" out to sit _beside_ "Usage" so that we can have nested pages here? Not sure...
3. If not, can you link to a heading in the same page in MkDocs? (like an `<a>` tag linking to an ID in HTML)

Admittedly, I've written a load more than what was in workflow.md but I hope that through being more comprehensive it will be a more useful document and _hopefully_ less confusing.

I've opted to avoid numbering any of the listed example setups - this is to hopefully future proof any references and/or links to the document should we add any further examples that would require rejiggling the order.

Thoughts?

**Comment by danbohea** [comment]
I've got my head around MkDocs' process & limitations for nesting pages and linking stuff together. I've opted for links to IDs within the same page (committed) as much of the nested file structure gets flattened in the sidebar menu, making documentation involving many files look rather daunting. This is likely just a limitation of the the theme though - maybe one day we can customise this to suit? One day! :D
